monai/transforms/utils_pytorch_numpy_unification.py

```diff
@@ -50,3 +51,17 @@ def in1d(x, y):
     if isinstance(x, np.ndarray):
         return np.in1d(x, y)
     return (x[..., None] == torch.tensor(y, device=x.device)).any(-1).view(-1)
+
+
+def where(condition: NdarrayOrTensor, x, y) -> NdarrayOrTensor:
+    """
+    Note that `torch.where` may convert y.dtype to x.dtype.
+    """
+    result: NdarrayOrTensor
+    if isinstance(condition, np.ndarray):
+        result = np.where(condition, x, y)
+    else:
+        x = torch.as_tensor(x, device=condition.device)
+        y = torch.as_tensor(y, device=condition.device, dtype=x.dtype)
```

Comment:
Hi @rijobro I added a `dtype` casting here to avoid an error on windows when x is torch.int32, y is torch.int64